MapboxMobileEvents/MMEAPIClient.m

```diff
@@ -31,30 +36,39 @@ - (instancetype)initWithAccessToken:(NSString *)accessToken userAgentBase:(NSStr
     return self;
 }
 
+- (void)reconfigure:(MMEEventsConfiguration *)configuration {
+    [self.sessionWrapper reconfigure:configuration];
+}
+
 - (void)postEvents:(NSArray *)events completionHandler:(nullable void (^)(NSError * _Nullable error))completionHandler {
     NSURLRequest *request = [self requestForEvents:events];
     [self.sessionWrapper processRequest:request completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
         NSError *statusError = nil;
-        NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
-        if (httpResponse.statusCode >= 400) {
-            NSString *descriptionFormat = @"The session data task failed. Original request was: %@";
-            NSString *reasonFormat = @"The status code was %ld";
-            NSString *description = [NSString stringWithFormat:descriptionFormat, request];
-            NSString *reason = [NSString stringWithFormat:reasonFormat, (long)httpResponse.statusCode];
-            NSDictionary *userInfo = @{NSLocalizedDescriptionKey: description,
-                                       NSLocalizedFailureReasonErrorKey: reason};
-            statusError = [NSError errorWithDomain:MMEErrorDomain code:1 userInfo:userInfo];
-        }
-        if (completionHandler) {
-            error = error ?: statusError;
-            completionHandler(error);
+        if ([response isKindOfClass:[NSHTTPURLResponse class]]) {
+            NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
+            statusError = [self statusErrorFromRequest:request andHTTPResponse:httpResponse];
+            if (completionHandler) {
```

Comment:
This `if` block can move below the parent `if`, and then you can remove the duplicate one below.